File: 8stringToInteger(atoi)/8stringToInteger(atoi).js
/**
 * @param {string} str
 * @return {number}
 */
var myAtoi = function(str) {
    let result = str.trim().match(/^[+-]?\d+/)
    result = BigInt(result ? result[0] : 0)
    return [result, -1n << 31n, (1n << 31n) - 1n].sort((a,b) => a - b)[1]
};

File: 8stringToInteger(atoi)/8stringToInteger(atoi).java
class Solution {
    public int myAtoi(String str) {
        final java.util.regex.Matcher matcher = java.util.regex.Pattern.compile("^[+-]?\\d+").matcher(str.trim());
        final java.math.BigInteger result = new java.math.BigInteger(matcher.find() ? matcher.group() : "0");
        return java.util.stream.Stream.of(result, java.math.BigInteger.valueOf(Integer.MAX_VALUE), java.math.BigInteger.valueOf(Integer.MIN_VALUE)).sorted().skip(1).findFirst().orElse(java.math.BigInteger.ZERO).intValue();
    }
}
